src/hdf5_time_series_storage.jl

```diff
@@ -300,24 +330,122 @@ function deserialize_time_series(
         path = _get_time_series_path(root, uuid)
         attributes = _read_time_series_attributes(storage, path, rows, T)
         @assert attributes["type"] == T
-        @assert length(attributes["dataset_size"]) == 2
         @debug "deserializing a Forecast" T
-        data = SortedDict{Dates.DateTime, Array}()
-        initial_timestamp = attributes["start_time"]
-        interval = attributes["interval"]
-        start_time = initial_timestamp + interval * (columns.start - 1)
-        if length(columns) == 1
-            data[start_time] = path["data"][rows, columns.start]
-        else
-            data_read = path["data"][rows, columns]
-            for (i, it) in
-                enumerate(range(start_time; length = length(columns), step = interval))
-                data[it] = @view data_read[1:length(rows), i]
-            end
+        data_type = attributes["data_type"]
+        data = get_hdf_array(path, data_type, attributes, rows, columns)
+        new_ts = T(ts_metadata, data)
+    end
+end
+
+function get_hdf_array(
+    path::HDF5.HDF5Group,
+    type::Type{<:CONSTANT},
```

Comment:
I don't see `type` being used.
```suggestion
    ::Type{<:CONSTANT},
```